tools/release_packages

```diff
@@ -0,0 +1,372 @@
+                prev = line.split('=')[-1].strip().replace('"', '')
+                new_file.write(line.replace(prev, new_version))
+            else:
+                new_file.write(line)
+
+    os.remove(init_file)
+    shutil.move(temp_abs_path, init_file)
+
+    message = 'Incremented __version__ in {fp} to {version}.'.format(
+        fp=init_file, version=new_version)
+    print(crayons.white(message, bold=True))
+
+def push_release_tag(version, upstream_or_origin):
+    print(crayons.yellow('Pushing release tag {}...'.format(version), bold=True))
+    git.tag(version)
+    git.push(upstream_or_origin, version)
+
+def commit_and_push_version_changes(version, init_files, upstream_or_origin):
+    message = 'Committing version number changes...{}'.format(version)
+    print(crayons.yellow(message, bold=True))
+
+    if upstream_or_origin == 'origin':
+        branch = 'release-{}'.format(version)
+        git.checkout('-b', branch)
+    else:
+        branch = 'master'
+        git.checkout(branch)
+
+    print(crayons.yellow('Diff:'))
+    print(git.diff())
+
+    git.add(*init_files)
+    commit_msg = 'python-zulip-api: Upgrade package versions to {}.'.format(
+        version)
+    print(crayons.yellow('Commit message: {}'.format(commit_msg), bold=True))
+    git.commit('-m', commit_msg)
+
+    git.push(upstream_or_origin, branch)
+
+def update_requirements_in_zulip_repo(zulip_repo_dir, version, hash_or_tag):
+    common = os.path.join(zulip_repo_dir, 'requirements', 'common.txt')
+    prod_lock = os.path.join(zulip_repo_dir, 'requirements', 'prod_lock.txt')
+    dev_lock = os.path.join(zulip_repo_dir, 'requirements', 'dev_lock.txt')
+    version_py = os.path.join(zulip_repo_dir, 'version.py')
+
+    def _edit_reqs_file(reqs, zulip_bots_line, zulip_line):
+        fh, temp_abs_path = tempfile.mkstemp()
+        with os.fdopen(fh, 'w') as new_file, open(reqs) as old_file:
+            for line in old_file:
+                if 'python-zulip-api' in line and 'zulip==' in line:
+                    new_file.write(zulip_line)
+                elif 'python-zulip-api' in line and 'zulip_bots' in line:
+                    new_file.write(zulip_bots_line)
+                else:
+                    new_file.write(line)
+
+        os.remove(reqs)
+        shutil.move(temp_abs_path, reqs)
+
+    url_zulip = 'git+https://github.com/zulip/python-zulip-api.git@{tag}#egg={name}=={version}_git&subdirectory={name}\n'
+    url_zulip_bots = 'git+https://github.com/zulip/python-zulip-api.git@{tag}#egg={name}=={version}+git&subdirectory={name}\n'
+
+    _edit_reqs_file(
+        prod_lock,
+        url_zulip_bots.format(tag=hash_or_tag, name='zulip_bots', version=version),
+        url_zulip.format(tag=hash_or_tag, name='zulip', version=version),
+    )
+    _edit_reqs_file(
+        dev_lock,
+        url_zulip_bots.format(tag=hash_or_tag, name='zulip_bots', version=version),
+        url_zulip.format(tag=hash_or_tag, name='zulip', version=version),
+    )
```

Comment:
Do a `map` for both `_edi_reqs_file` calls?

Reply:
Agreed, much more concise! Thanks, Robert! :)